### PR TITLE
Add conditional includes support to issue-config files

### PR DESCRIPTION
**Pull request by kkaarreell**
Extends the issue-config include mechanism to support conditional includes with a 'when' clause that evaluates conditions before loading resources. This enables dynamic configuration based on event type, artifact properties, and CLI parameters.

Changes:
- Support both simple string format and dict format with optional 'when' condition in include directives
- Pass event-specific variables (EVENT, ERRATUM, COMPOSE, ROG) and CLI variables (CONTEXT, ENVIRONMENT) to conditional evaluations
- Maintain backward compatibility with existing string-only includes
- Add comprehensive tests for all variable types and edge cases
- Update documentation with examples and available variables

Available variables in conditions:
- EVENT: The event object
- ERRATUM: Erratum artifact (when applicable)
- COMPOSE: Compose artifact (when applicable)
- ROG: RoG artifact (when applicable)
- CONTEXT: CLI context parameters
- ENVIRONMENT: CLI environment variables

Example usage:
  include:
    - url: path/to/config.yaml when: COMPOSE and COMPOSE.id is match('CentOS-Stream-.*')

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add support for conditional includes in issue-config files, allowing included configurations to be loaded based on evaluated conditions and CLI/runtime context.

Enhancements:
- Extend issue-config include handling to accept either simple string entries or dicts with url and optional when condition, evaluating conditions with supplied variables before loading child configs.
- Wire Jira CLI command to pass event, artifact, context, and environment variables into issue-config loading so conditional includes can react to runtime data.

Documentation:
- Document simple and conditional include formats for issue-config files, including available condition variables and example configurations.

Tests:
- Add unit tests and fixture configs covering conditional include behavior, including enabled/disabled flags, artifact-based conditions, CLI context/environment conditions, multiple matching includes, and no-match scenarios.

### CODE REVIEW

File: tests/unit/test_include.py [hunk 1]
```diff
@@ -49,3 +49,272 @@ def test_recipe_include():
...
+def test_issue_config_conditional_include_enabled():
+    # Load config with ENABLE_FEATURE set to True
+    config = IssueConfig.read_file(
+        'tests/unit/data/issue-config-conditional-parent.yaml',
+        variables={'ENABLE_FEATURE': True})
+
+    # Should have 2 issues: 1 from parent + 1 from child1 (always included)
+    assert len(config.issues) == 2
+
+    # Project comes from parent
```
[the remark this answers is not in the document]
Comment (kkaarreell) [reply tests/unit/test_include.py:65]: fixed
